src/version.c

```diff
@@ -32,6 +32,59 @@
 #include "signature.h"
 #include "swupd.h"
 
+#define MAX_VERSION_STR_SIZE 11
+
+int get_server_format(int server_version)
+{
+	if (server_version < 0) {
+		return -1;
+	}
+
+	int ret, err = -1;
+	char *url;
+
+	char format_string[MAX_VERSION_STR_SIZE];
+	struct curl_file_data format = {
+		MAX_VERSION_STR_SIZE, 0,
+		format_string
+	};
+
+	string_or_die(&url, "%s/%d/format", globals.version_url, server_version);
+	ret = swupd_curl_get_file_memory(url, &format);
+	free_string(&url);
+
+	if (ret) {
+		return -1;
+	}
+
+	format.data[format.len] = '\0';
+	err = strtoi_err(format.data, &ret);
+	if (err != 0) {
+		return -1;
+	}
+
+	return ret;
+}
+
+int get_current_format()
+{
+	char *temp_format_buffer = NULL;
+	int ret, err = -1;
+	ret = get_value_from_path(&temp_format_buffer, DEFAULT_FORMAT_PATH, false);
```

Comment:
I think we need to support a path prefix set by the user the  using the -p flag

Reply:
Isnt it doing the same ? I thought sending a false as the last argument means it is not an absolute path and would take one by prefixing the state dir path

Reply:
ooh, you are right, I missed that 😄